runner/local/output.go

```diff
@@ -35,17 +68,92 @@ func (s *localOutputCreator) Create(id string) (runner.Output, error) {
 		return nil, err
 	}
 	// We don't need a / between hostname and path because absolute paths start with /
-	return &localOutput{f: f, hostname: s.hostname, absPath: absPath}, nil
+	uri := fmt.Sprintf("file://%s%s", s.hostname, absPath)
+	if s.httpUri != "" {
+		uri = fmt.Sprintf("%s/%s?file=%s", s.httpUri, id, uri)
+		s.pathMap[strings.Trim(id, "/")] = absPath
+		s.pathMap[filepath.Base(absPath)] = absPath
+	}
+	return &localOutput{f: f, absPath: absPath, uri: uri}, nil
+}
+
+// Serves a minimal page that does ajax log tailing of the specified path
+// When '?content=true' is specified, this serves the content directly without ajax.
+// Does not check the request path, either it finds the local file or 404s.
+func (s *localOutputCreator) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	clientHtml :=
+		`<html>
+<script type="text/javascript">
+  var prevLength = 0
+  checkAtBottom = function() {
+    //scrolling: http://stackoverflow.com/a/22394544
+    var scrollTop = (document.documentElement && document.documentElement.scrollTop) || document.body.scrollTop;
+    var scrollHeight = (document.documentElement && document.documentElement.scrollHeight) || document.body.scrollHeight;
+    return (scrollTop + window.innerHeight) >= scrollHeight;
+  }
+  gotoBottom = function() {
+    var scrollHeight = (document.documentElement && document.documentElement.scrollHeight) || document.body.scrollHeight;
+    var scrollLeft = (document.documentElement && document.documentElement.scrollLeft) || document.body.scrollLeft;
+    window.scrollTo(scrollLeft, scrollHeight);
+  }
+  sendRequest = function() {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function () {
+      var DONE=4, OK=200;
+      if (xhr.readyState === DONE && xhr.status == OK) {
+        var wasAtBottom = checkAtBottom()
+        document.body.innerText += xhr.responseText.substring(prevLength);
+        if (wasAtBottom)
+          gotoBottom()
+        prevLength = xhr.responseText.length
+      }
+    }
+    //TODO: request range to get delta.
+    xhr.open("GET", location.href + (location.search=="" ? "?" : "&") + "content=true");
+    xhr.send();
+  };
+  sendRequest()
+  setInterval(sendRequest, 2000)
+</script>
+</html>
+`
+	if strings.TrimSuffix(r.URL.Path, "/")+"/" == s.HttpPath() {
+		http.StripPrefix(s.HttpPath(), http.FileServer(http.Dir(s.tmp.Dir))).ServeHTTP(w, r)
+		return
+	}
+	path := strings.TrimPrefix(r.URL.Path, s.HttpPath())
+	filepath, ok := s.pathMap[path]
+	if !ok {
+		http.Error(w, "Unrecognized path", http.StatusNotFound)
```

Comment:
https://golang.org/pkg/net/http/#ServeFile

Oh, but you want to do this JS thing to keep reloading, huh?